src/smk/viewer-leaflet/viewer-leaflet.js

```diff
@@ -177,44 +183,38 @@ include.module( 'viewer-leaflet', [ 'viewer', 'leaflet', 'layer-leaflet', /*'fea
     ViewerLeaflet.prototype.basemap.Streets.create = createBasemapEsri
 
     ViewerLeaflet.prototype.basemap.Imagery.create = createBasemapEsri
-    ViewerLeaflet.prototype.basemap.Imagery.labels = [ 'ImageryTransportation', 'ImageryLabels' ]
 
     ViewerLeaflet.prototype.basemap.Oceans.create = createBasemapEsri
-    ViewerLeaflet.prototype.basemap.Oceans.labels = [ 'OceansLabels' ]
-
-    ViewerLeaflet.prototype.basemap.NationalGeographic.create = createBasemapEsri
 
     ViewerLeaflet.prototype.basemap.ShadedRelief.create = createBasemapEsri
-    ViewerLeaflet.prototype.basemap.ShadedRelief.labels = [ 'ShadedReliefLabels' ]
 
     ViewerLeaflet.prototype.basemap.DarkGray.create = createBasemapEsri
-    ViewerLeaflet.prototype.basemap.DarkGray.labels = [ 'DarkGrayLabels' ]
 
     ViewerLeaflet.prototype.basemap.Gray.create = createBasemapEsri
-    ViewerLeaflet.prototype.basemap.Gray.labels = [ 'GrayLabels' ]
-
+    
     ViewerLeaflet.prototype.basemap.StamenTonerLight.create = createBasemapTiled
     ViewerLeaflet.prototype.basemap.StamenTonerLight.url = 'https://stamen-tiles-{s}.a.ssl.fastly.net/toner-lite/{z}/{x}/{y}.png'
     ViewerLeaflet.prototype.basemap.StamenTonerLight.attribution = "Map tiles by <a href='http://stamen.com'>Stamen Design</a>, under <a href='http://creativecommons.org/licenses/by/3.0'>CC BY 3.0</a>. Data by <a href='http://openstreetmap.org'>OpenStreetMap</a>, under <a href='http://www.openstreetmap.org/copyright'>ODbL</a>."
 
-    function createBasemapEsri( id ) {
-        /* jshint -W040 */
-        var opt = Object.assign( { detectRetina: true }, this.option )
-
-        var lys = []
-        lys.push( L.esri.basemapLayer( id, opt ) )
+    function createBasemapEsri( apiId ) {
+        const apiKey = SMK.MAP[1].viewer.esriApiKey;
```

Comment:
This lookup is unsafe, SMK.MAP[1] doesn't always exist. The '1' is the 'smk-id' and can be set by the user. See the SMK/Initialization docs for details.

Reply:
I'll reinstate the passing around of the "esriApiKey" value that we get from the smk object in the initialization method.

Reply:
You have a few other options for accessing the property:
1. Object.getOwnPropertyNames
2. Object.keys
3. for const prop in my_object